### DISCUSSION TITLE
Receiving broadcast event data

### DISCUSSION THREAD

**Discussion by ekandreas** [livewire/livewire#6378] [7. Bugs (archived)]
### Livewire version

3.0-beta.9

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

1. Create an event with public properties
2. Dispatch the event
3. Listen to the event inside a Livewire component.
4. Try to resolve the data sent.

### Additional information

In version 2 it was possible to get the event data. But in version 3 there are no data sent from the event props.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by yoeriboven** [comment]
Same for me!

Here is a repo in which you can see the bug: https://github.com/yoeriboven/livewire-echo

<img width="1440" alt="Scherm­afbeelding 2023-08-17 om 17 22 52" src="https://github.com/livewire/livewire/assets/4047804/fcdb4ef5-28e9-4c04-90e1-69a0fa4f6814">

**Reply by ekandreas** [reply]
Try put another argument in your function.
From this:
```
    public function streamer($event) {
        $this->answer = $event["text"];
    }
```
to this:
```
    public function streamer(...$event) {
        $this->answer = $event[1];
    }
```
Perhaps there are some mapping missing here? But this is a workaround.

**Reply by yoeriboven** [reply]
Yep, that works! Would be nice if it could be fixed to use keys again though. :)

**Reply by ekandreas** [reply]
Agree on that but I'm to lazy for setting up a project and make a "proof of bug".
Thanks for supporting the case.

**Reply by adesugbaa** [reply]
This is a big difference from v2.

I am unable to use the presence callbacks cause the data is returned in a strange format.

The echo presence listeners are returning the data in a strange way.

- In the here callback, you need to access the data as a variable number of arguments instead of an array of users
- In the joining and leaving callback, the user is returned as an array (where you lose the keys) instead of a single object with keys

Is this a bug???

**Reply by ekandreas** [reply]
I think it's a bug but not sure Caleb has seen it. 
Perhaps make a proper issue report?

**Reply by edalzell** [reply]
even better would be a failing test

**Reply by yoeriboven** [reply]
PR'd here: https://github.com/livewire/livewire/pull/6413

**Reply by edalzell** [reply]
I've got that PR pulled and I still can't get presence stuff going, is there a good way to debug it?

**Reply by yoeriboven** [reply]
Sorry have never used those before so don't know anything about them.
